src/main/java/io/cdap/plugin/googleads/common/GoogleAdsBaseConfig.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.googleads.common;
+
+import com.google.api.ads.common.lib.exception.OAuthException;
+import com.google.api.ads.common.lib.exception.ValidationException;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.common.ReferencePluginConfig;
+
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+
+/**
+ * Provides all required configuration for reading Google AdWords reports
+ */
+public class GoogleAdsBaseConfig extends ReferencePluginConfig {
+
+  public static final String REFRESH_TOKEN = "refreshToken";
+  public static final String CLIENT_ID = "clientId";
+  public static final String CLIENT_SECRET = "clientSecret";
+  public static final String DEVELOPER_TOKEN = "developerToken";
+  public static final String CLIENT_CUSTOMER_ID = "clientCustomerId";
+  public static final String START_DATE = "startDate";
+  public static final String END_DATE = "endDate";
+  public static final String INCLUDE_REPORT_SUMMARY = "includeReportSummary";
+  public static final String USE_RAW_ENUM_VALUES = "useRawEnumValues";
+  public static final String INCLUDE_ZERO_IMPRESSIONS = "includeZeroImpressions";
+  @Name(REFRESH_TOKEN)
+  @Description("Authorization to download the report")
+  @Macro
+  public String refreshToken;
+  @Name(CLIENT_ID)
+  @Description("OAuth 2.0 client ID from console")
+  @Macro
+  public String clientId;
+  @Name(CLIENT_SECRET)
+  @Description("OAuth 2.0 client Secret from console")
+  @Macro
+  public String clientSecret;
+  @Name(DEVELOPER_TOKEN)
+  @Description("Developer token consisting of unique string")
+  @Macro
+  public String developerToken;
+  @Name(CLIENT_CUSTOMER_ID)
+  @Description("Customer ID of the client account")
+  @Macro
+  public String clientCustomerId;
+  @Name(INCLUDE_REPORT_SUMMARY)
+  @Description("Specifies whether report include a summary row containing the report totals.")
+  @Macro
+  public Boolean includeReportSummary;
+  @Name(USE_RAW_ENUM_VALUES)
+  @Description("Specifies whether returned format to be the actual enum value.")
```

Comment:
This is gramatically not correct also hard to undestand.

Maybe it should be:
"Specifies whether to fields in enum format."
or something like that.